src/yunohost/log.py

```diff
@@ -0,0 +1,424 @@
+                if 'log_path' in metadata:
+                    log_path = metadata['log_path']
+            except yaml.YAMLError:
+                error = m18n.n('log_corrupted_md_file', file=md_path)
+                if os.path.exists(log_path):
+                    logger.warning(error)
+                else:
+                    raise MoulinetteError(errno.EINVAL, error)
+
+    # Display logs if exist
+    if os.path.exists(log_path):
+        from yunohost.service import _tail
+        logs = _tail(log_path, int(number))
+        #logs = [{"datetime": x.split(": ", 1)[0].replace("_", " "), "line": x.split(": ", 1)[1]}  for x in logs if x]
+        infos['log_path'] = log_path
+        infos['logs'] = logs
+
+    return infos
+
+def is_unit_operation(entities='app,domain,service,user', exclude='auth,password', operation_key=None, auto=True):
+    """
+    Configure quickly a unit operation
+
+    This decorator help you to configure quickly the record of a unit operations.
+
+    Argument:
+    entities    A list seperated by coma of entity types related to the unit
+    operation. The entity type is searched inside argument's names of the
+    decorated function. If something match, the argument value is added as
+    related entity.
+
+    exclude     Remove some arguments from the context. By default, arguments
+    called 'password' and 'auth' are removed. If an argument is an object, you
+    need to exclude it or create manually the unit operation without this
+    decorator.
+
+    operation_key   Key describing the unit operation. If you want to display a
+    well formed description you should add a translation key like this
+    "log_" + operation_key in locales files.
+
+    auto        If true, start the recording. If False, the unit operation object
+    created is given to the decorated function as the first argument and you can
+    start recording at the good time.
+    """
+    def decorate(func):
+        def func_wrapper(*args, **kwargs):
+            # For a strange reason we can't use directly the arguments from
+            # is_unit_operation function. We need to store them in a var before.
+            entities_list = entities.split(',')
+            exclude_list = exclude.split(',')
+            op_key = operation_key
+            related_to = []
+
+            if op_key is None:
+                op_key = func.__name__
+
+            # In case the function is called directly from an other part of the
+            # code
+            if len(args) > 0:
+                from inspect import getargspec
+                keys = getargspec(func).args
+                if 'uo' in keys:
+                    keys.remove('uo')
+                for k, arg in enumerate(args):
+                    kwargs[keys[k]] = arg
+                args = ()
+
+            # Search related entity in arguments of the decorated function
+            for entity in entities_list:
+                entity = entity.split(':')
```

Comment:
Again, we can have a list of string or a list of tuple, that seems weird to invent a format store in a string :confused:

Reply:
If you keep that you probably want to do a `.split(":", 1)`